scripts/get_test_metrics/get_test_metrics.sh

```diff
@@ -0,0 +1,252 @@
+  exit 1
+}
+
+parse_params() {
+  # default values of variables set from params
+  project_name=pn
+  work_dir=$HOME/tmp/deploy
+  aws_profile=""
+  aws_region=""
+  env_type=""
+
+
+  while :; do
+    case "${1-}" in
+    -h | --help) usage ;;
+    -v | --verbose) set -x ;;
+    -p | --profile) 
+      aws_profile="${2-}"
+      shift
+      ;;
+    -r | --region) 
+      aws_region="${2-}"
+      shift
+      ;;
+    -w | --work-dir) 
+      work_dir="${2-}"
+      shift
+      ;;
+    -?*) die "Unknown option: $1" ;;
+    *) break ;;
+    esac
+    shift
+  done
+
+  args=("$@")
+
+  # check required params and arguments
+  [[ -z "${aws_region-}" ]] && usage
+  return 0
+}
+
+dump_params(){
+  echo ""
+  echo "######      PARAMETERS      ######"
+  echo "##################################"
+  echo "Project Name:       ${project_name}"
+  echo "Work directory:     ${work_dir}"
+  echo "AWS region:         ${aws_region}"
+  echo "AWS profile:        ${aws_profile}"
+}
+
+# START SCRIPT
+
+parse_params "$@"
+dump_params
+
+
+echo ""
+echo "=== Base AWS command parameters"
+aws_command_base_args=""
+if ( [ ! -z "${aws_profile}" ] ) then
+  aws_command_base_args="${aws_command_base_args} --profile $aws_profile"
+fi
+if ( [ ! -z "${aws_region}" ] ) then
+  aws_command_base_args="${aws_command_base_args} --region  $aws_region"
+fi
+echo ${aws_command_base_args}
+
+echo -n "insert start time: "
+read start_time
+
+echo -n "insert end time: "
+read end_time
+
+
+dir=monitoring_${aws_profile}_$(date +"%d-%m-%Y")
+
+mkdir -p $dir
+
+echo "change work directory"
+
+cd $dir
+
+echo time is $start_time $end_time
+
+echo "start export CPU  metrics for ECS"
+
+echo "==> start export CPU  metrics for ECS"
+
+for cpumetrics in  $(aws ${aws_command_base_args} cloudwatch list-metrics --metric-name CPUUtilization --namespace AWS/ECS  --output text | grep ServiceName | awk '{print $2",Value="$3}') ; do
+aws ${aws_command_base_args} cloudwatch get-metric-statistics --metric-name CPUUtilization  --start-time $start_time --end-time $end_time --period 60 --statistics Maximum --dimensions Name=$cpumetrics Name=ClusterName,Value=pn-core-ecs-cluster --namespace AWS/ECS >> cpu_metrics_$(echo $cpumetrics  | cut -d "=" -f 2).json && echo  cpu metrics export for: $( echo $cpumetrics  | cut -d "=" -f 2); 
+done
+
+echo "==> cpu metrics done"
+echo "==> start export RAM metrics for ECS"
+
+for rammetrics in  $(aws ${aws_command_base_args} cloudwatch list-metrics --metric-name MemoryUtilization --namespace AWS/ECS   --output text | grep ServiceName | awk '{print $2",Value="$3}') ; do
+aws ${aws_command_base_args} cloudwatch get-metric-statistics --metric-name MemoryUtilization  --start-time $start_time --end-time $end_time --period 60 --statistics Maximum --dimensions Name=$rammetrics Name=ClusterName,Value=pn-core-ecs-cluster --namespace AWS/ECS >> ram_metrics_$(echo $rammetrics  | cut -d "=" -f 2).json && echo  ram metrics export for: $( echo $rammetrics  | cut -d "=" -f 2);
```

Comment:
@mbrachi potrebbe essere utile riceve in input il period (opzionale, di default a 60)

Reply:
implementato, lo gestiamo mediante parametro opzionale "-P" allo start dello script. Nel caso in cui non venga inserito, lo script viene eseguito con period=60